lib/fog/identity/openstack/v2.rb

```diff
@@ -169,8 +169,17 @@ def credentials
         end
 
         class Real < Fog::Identity::OpenStack::Real
+          def initialize(args)
+            @path_prefix = if args[:no_path_prefix]
+                             ''
+                           else
+                             'v3'
+                           end
+            super
+          end
+
           def default_path_prefix
-            'v2.0'
+            @path_prefix
```

Comment:
What do you mean? The @path_prefix is set to its respective version value (v3 or v2.0), if a the path contains a version then we remove the prefix because the path has it hard coded (so to speak).

Reply:
I mean that the ``initialize`` method sets ``@path_prefix`` to ``'v3'`` for this class, so this effectively changes the value returned by ``default_path_prefix`` in the case where ``args[:no_path_prefix]`` is falsey (the default, I presume) from ``'v2.0'`` to ``'v3'`` for this class.